### ISSUE TITLE
Speed up rendering of Pages / Have them all render at once.

### ISSUE THREAD

**Issue by WillHaack** [ByunLab/staRt-ionic#391]
Right now many pages on the app look choppy when they render. We should try to see if there are anyways we can speed up the rendering process, and if not, we should perhaps have an element (with maybe a beach themed loading icon) that covers an entire page stay on top, and then vanish once the page is done loading.

**Comment by WillHaack** [comment]
From Slack:

@jonchin When you decide to tackle https://github.com/ByunLab/staRt-ionic/issues/391  , I would suggest that the first piece you look at is the loading of the "recordings" tab on the profiles page. For me it sometimes takes forever to load the list of recordings. It seems as though some time expensive process is running, such as perhaps the content of the audio files being read from disk into memory.

**Comment by jonathan-chin** [comment]
hmm. any chance I can get access to your account where loading recordings takes forever? are there a lot of recordings and/or are the recordings long?

**Comment by WillHaack** [comment]
@jonathan-chin I did have a lot of recordings, some short and some long, but I deleted them. Also, I think the actual recordings are stored locally on the device, so I don't think you would be able to access them from logging in from another simulator/iPad. As a side note, we need to test recovering an account from another device.

**Comment by jonathan-chin** [comment]
hey @WillHaack I did some investigating. I couldn't reproduce your lag on the recordings list as much as I tried. Here are my notes for each page:

```
Profiles page
	Takes a while to load registered participants
	removing css transitions helps
	I got no slow down once on the page, switching through various profiles / recordings / etc

Free Play page
	Experience lag as the wave and sign posts fall into place
	I think the lagging elements are generated by the LPC directive

Quest Page
	Similar lag as to Free Play but much more pronounced
	More images need to load

Quiz Page
	A little lag for images to load
	Very long lag for first word to show up
	Sometimes first word doesn't show up at all, but will operate as normal after "blind" rating once
```

I think the Quiz / Quest / Free Play lag issues are due to the order of how we build the canvas during in the LPC. I tried to dig around but am not familiar with that part of the app. Maybe @helencarey might have insight?

Can we switch up the order of elements rendering in the LPC? My gut says that everything is blocked until the **math** of the wave is set up. Could we:

1. Load the less computationally difficult items first
2. Load a dummy / static wave (even if it's just an image) until the wave math is done?

I also found an inconsistent bug with long word quizzes (and possibly other quizzes). I'll open a separate ticket for that.

**Comment by WillHaack** [comment]
@jonathan-chin 

Thanks for writing all of these up. I have experienced these bugs myself, as I'm sure everyone else has. I agree that the lag comes from setting up the LPC wave. I'm not sure what you mean by "the **math** of the wave", my intuition is the delay comes from the AudioPlugin taking a second to allocate memory for the sound wave / connect to the microphone. In any case, there is some function that hangs for a second. I agree we should load whatever we can before that hanging function, including a static / dummy wave.

Regarding the slow loading of the recordings list, it may be because I was having a connection issue here. It also may have come from a change that I made in the branch I was working on. Don't worry about it for now. The issues you've stated here and the first-word-not-loading bug are more important.

**Comment by jonathan-chin** [comment]
I think I use the term "math of the wave" to refer to whatever the inner workings are. The audio plugin is, for most intents and purposes, a black box to me.

I think that certain assets are drawn by the plugin that we could instead load through HTML. also, does the audio plugin fire off an event when it is fully loaded? if it does, we can use that to load/unload a dummy wave.
